build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/ArtifactoryClientConfiguration.java

```diff
@@ -226,24 +240,132 @@ public Log getLog() {
+        return secretKey;
+    }
+
     public void persistToPropertiesFile() {
         if (StringUtils.isEmpty(getPropertiesFile())) {
             return;
         }
+        try (FileOutputStream fos = new FileOutputStream(new File(getPropertiesFile()).getCanonicalFile())) {
+            preparePropertiesToPersist().store(fos, "BuildInfo configuration property file");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public byte[] persistToEncryptedPropertiesFile(OutputStream os) throws IOException {
+        if (StringUtils.isEmpty(getPropertiesFile())) {
+            return null;
+        }
+        return encryptedPropertiesToFile(os, preparePropertiesToPersist());
+    }
+
+    private Properties preparePropertiesToPersist() {
         Properties props = new Properties();
-        props.putAll(filterMapNullValues(root.props));
         props.putAll(filterMapNullValues(rootConfig.props));
-        FileOutputStream fos = null;
+        props.putAll(filterMapNullValues(root.props));
+
+        // Properties that have the 'artifactory.' prefix are deprecated.
+        // For backward compatibility reasons, both will be added to the props map.
+        // The build-info 2.32.3 is the last version to be using the deprecated properties as its primary.
+        Properties artifactoryProps = new Properties();
+        for (String key : props.stringPropertyNames()) {
+            artifactoryProps.put("artifactory." + key, props.getProperty(key));
+        }
+        props.putAll(artifactoryProps);
+
+        return props;
+    }
+
+    /**
+     * Decrypts properties from a file using the provided secret key.
+     *
+     * @param filePath  The path to the file containing encrypted properties.
+     * @param secretKey The secret key used for decryption.
+     * @return A Properties object containing the decrypted properties.
+     */
+    public static Properties decryptPropertiesFromFile(String filePath, byte[] secretKey) throws IOException {
+        return decryptProperties(FileUtils.readFileToByteArray(new File(filePath)), secretKey);
+    }
+
+    /**
+     * Encrypts properties into a byte array using the provided secret key.
+     *
+     * @param properties The Properties object to be encrypted.
+     * @param secretKey  The secret key used for encryption.
+     * @return A byte array representing the encrypted properties.
+     */
+    private static byte[] encryptProperties(Properties properties, byte[] secretKey) {
         try {
-            fos = new FileOutputStream(new File(getPropertiesFile()).getCanonicalFile());
-            props.store(fos, "BuildInfo configuration property file");
-        } catch (IOException e) {
+            Cipher cipher = Cipher.getInstance(TRANSFORMATION);
+            SecretKeySpec secretKeySpec = new SecretKeySpec(secretKey, ALGORITHM);
+            cipher.init(Cipher.ENCRYPT_MODE, secretKeySpec);
```

Comment:
Let's use CBC algorithm since it is considered safer than ECB.
See this example:
https://www.baeldung.com/java-aes-encryption-decryption#1-string